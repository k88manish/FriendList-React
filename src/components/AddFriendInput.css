:local(.addFriendInput) {
  border-radius: 0;
  border-color: #abaaaa;
  border-left: 0;
  border-right: 0;
}

:local(.inputContainer) {
  display: flex;
  flex-direction: row;
}

:local(.selectInput) {
  border-radius: 0;
  border-color: #abaaaa;
  border-left: 0;
  border-right: 0;
  width: 45%;
}
