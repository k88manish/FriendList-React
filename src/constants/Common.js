export const GENDER = ["Male", "Female"];
export const PREVIOUS = "Prev";
export const NEXT = "Next";
export const MALE = "Male";
