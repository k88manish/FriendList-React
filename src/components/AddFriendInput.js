import React, { Component } from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import { GENDER } from "../constants/Common";
import styles from "./AddFriendInput.css";

class AddFriendInput extends Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      name: this.props.name || "",
      gender: this.props.gender || "Male"
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSelectChange = this.handleSelectChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  render() {
    return (
      <div className={styles.inputContainer}>
        <select className={classnames("form-control", styles.selectInput)} onChange={this.handleSelectChange}>
          {GENDER.map(e => (
            <option key={e} value={e}>
              {e}
            </option>
          ))}
        </select>
        <input
          type="text"
          autoFocus="true"
          name="name"
          className={classnames("form-control", styles.addFriendInput)}
          placeholder="Type the name of a friend"
          value={this.state.name}
          onChange={this.handleChange}
          onKeyDown={this.handleSubmit}
        />
      </div>
    );
  }

  handleChange(e) {
    this.setState({ name: e.target.value });
  }

  handleSelectChange(e) {
    this.setState({ gender: e.target.value });
  }

  handleSubmit(e) {
    if (e.which === 13) {
      const { name } = this.state;
      if (name.trim().length === 0) {
        alert("Name can not be blank.");
        return;
      }
      this.props.addFriend(name, this.state.gender);
      this.setState({ name: "", gender: "Male" });
    }
  }
}

AddFriendInput.propTypes = {
  addFriend: PropTypes.func.isRequired
};

export default AddFriendInput;
