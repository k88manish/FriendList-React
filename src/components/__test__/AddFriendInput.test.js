import React from "react";
import { configure } from "enzyme";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-15";
import AddFriendInput from "../AddFriendInput";
import { MALE } from "../../constants/Common";

configure({ adapter: new Adapter() });

const mockAddFriend = jest.fn();
const propsData = {
  addFriend: mockAddFriend
};

describe("Add Input Test Suits", () => {
  it("it should render correctly", () => {
    const componentWrapper = shallow(<AddFriendInput {...propsData} />);
    expect(componentWrapper).toMatchSnapshot();
  });

  it("simulate submit on empty name field in form", () => {
    const componentWrapper = shallow(<AddFriendInput {...propsData} />);
    componentWrapper.setState({ name: "", gender: MALE });
    expect(mockAddFriend).toHaveBeenCalledTimes(0);
    componentWrapper.find({ name: "name" }).simulate("keydown", { which: 13 });
    expect(mockAddFriend).toHaveBeenCalledTimes(0);
  });

  it("simulate submit on complete form", () => {
    const componentWrapper = shallow(<AddFriendInput {...propsData} />);
    componentWrapper.setState({ name: "Manish", gender: MALE });
    componentWrapper.find({ name: "name" }).simulate("keydown", { which: 13 });
    expect(mockAddFriend).toHaveBeenCalledTimes(1);
  });
});
