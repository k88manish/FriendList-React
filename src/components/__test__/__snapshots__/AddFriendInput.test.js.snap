exports[`Add Input Test Suits it should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddFriendInput
    addFriend={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <select
          className="form-control"
          onChange={[Function]}>
          <option
                    value="Male">
                    Male
          </option>
          <option
                    value="Female">
                    Female
          </option>
</select>,
        <input
          autoFocus="true"
          className="form-control"
          name="name"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Type the name of a friend"
          type="text"
          value="" />,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="Male">
              Male
</option>,
            <option
              value="Female">
              Female
</option>,
          ],
          "className": "form-control",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Male",
            "nodeType": "host",
            "props": Object {
              "children": "Male",
              "value": "Male",
            },
            "ref": null,
            "rendered": "Male",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "Female",
            "nodeType": "host",
            "props": Object {
              "children": "Female",
              "value": "Female",
            },
            "ref": null,
            "rendered": "Female",
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoFocus": "true",
          "className": "form-control",
          "name": "name",
          "onChange": [Function],
          "onKeyDown": [Function],
          "placeholder": "Type the name of a friend",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <select
            className="form-control"
            onChange={[Function]}>
            <option
                        value="Male">
                        Male
            </option>
            <option
                        value="Female">
                        Female
            </option>
</select>,
          <input
            autoFocus="true"
            className="form-control"
            name="name"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Type the name of a friend"
            type="text"
            value="" />,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="Male">
                Male
</option>,
              <option
                value="Female">
                Female
</option>,
            ],
            "className": "form-control",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Male",
              "nodeType": "host",
              "props": Object {
                "children": "Male",
                "value": "Male",
              },
              "ref": null,
              "rendered": "Male",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "Female",
              "nodeType": "host",
              "props": Object {
                "children": "Female",
                "value": "Female",
              },
              "ref": null,
              "rendered": "Female",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoFocus": "true",
            "className": "form-control",
            "name": "name",
            "onChange": [Function],
            "onKeyDown": [Function],
            "placeholder": "Type the name of a friend",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
